### PR TITLE
Detect empty links on markdown rendering and issue an error

### PR DESCRIPTION
**Pull request by zbrox**
Referencing #871

Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [X] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [ ] Have you created/updated the relevant documentation page(s)?

### CODE REVIEW

File: components/rendering/src/markdown.rs [hunk 1]
```diff
@@ -242,6 +242,10 @@ pub fn markdown_to_html(content: &str, context: &RenderContext) -> Result<Render
 
                         Event::Start(Tag::Image(link_type, src, title))
                     }
+                    Event::Start(Tag::Link(link_type, link, title)) if link.is_empty() => {
+                        error = Some(Error::msg("There is a link that is missing a URL"));
```
Comment (Keats) [inline components/rendering/src/markdown.rs:246]: Can we add the link type/title to the error message?

Comment (zbrox) [reply components/rendering/src/markdown.rs:246]: I actually had it with the title in the message initially but for some reason the `title` variable there is an empty string. I didn't have more time to look into it. My initial guess was that maybe the Markdown parser is not properly parsing a link without a URL.

Comment (Keats) [reply components/rendering/src/markdown.rs:246]: ah interesting, I'll see if I can reproduce it and open an issue on pulldown-cmark

**Comment by Keats** [comment]
A bit late but can you add a test for that?

**Comment by zbrox** [comment]
> A bit late but can you add a test for that?

Yeah, sure. I'll look into that.

File: components/rendering/src/markdown.rs (hunk 1 above)
Comment (zbrox) [reply components/rendering/src/markdown.rs:246]: @Keats did you manage to reproduce it? Is it really an issue or did I misunderstand something?

File: components/rendering/tests/markdown.rs [hunk 2]
```diff
@@ -852,3 +852,24 @@ fn leaves_custom_url_scheme_untouched() {
...
+        &config,
+        "https://vincent.is/",
+        &permalinks_ctx,
+        InsertAnchor::None,
+    );
+
+    let res = render_content(content, &context);
+
+    assert!(res.is_err());
+}
```
Comment (zbrox) [inline components/rendering/tests/markdown.rs:878]: @Keats I was not terribly creative with the test. Should I test for something else other than erring out?

File: components/rendering/src/markdown.rs (hunk 1 above)
Comment (Keats) [reply components/rendering/src/markdown.rs:246]: I haven't tried yet, I'll check when I have time

File: components/rendering/tests/markdown.rs (hunk 2 above)
Comment (Keats) [reply components/rendering/tests/markdown.rs:878]: You can check that the description contains the expected string

**Comment by Keats** [comment]
Thanks!

File: components/rendering/src/markdown.rs (hunk 1 above)
Comment (Keats) [reply components/rendering/src/markdown.rs:246]: Ah nevermind, `title` is `[I'm an inline-style link with title](https://www.google.com "Google's Homepage")` Google's Homepage. It looks like you don't get the link text in the event

**Pull request merged**